worker/upload.go

```diff
@@ -543,40 +539,26 @@ func (mgr *uploadManager) Upload(ctx context.Context, r io.Reader, contracts []a
 				case <-ctx.Done():
 				}
 				return
-			} else if up.packing && errors.Is(err, io.ErrUnexpectedEOF) {
-				mem.Release()
-
-				// uploadPacking is true, we return the partial slab without
-				// uploading.
-				partialSlab = data[:length]
+			} else if errors.Is(err, io.ErrUnexpectedEOF) && up.packing {
+				go mgr.addPartialSlab(ctx, mem, up.rs, data[:length], up.contractSet, respIndex, respChan)
```

Comment:
There are 3 issues with this.
1. the partial slab is the last slab, so there is no need to spin up a new goroutine for this
2. adding the partial slab to the bus here means that we add it to the buffer before knowing whether the upload succeeds, possibly wasting space in the buffer.
3. this hides where `mem.Release` is called which isn't great. Similar to how a `Close` of a file should be close to where you open it.

Reply:
These are all valid points and I considered all of them to be a tradeoff in favour of removing the special treatment of partial slabs. I reverted it though, it's not that I didn't think of the issues you mentioned here, for me it improved readability and got rid of some nasty little things like decrementing the slabIndex.